### PR TITLE
Add new RAG tutorial

### PR DESCRIPTION
**Pull request by dwdougherty**
Hi @ollopaci and @ViktarStarastsenka. Here's a new tutorial for RAG that was requested by Michelle (via Yiftach). Michelle has reviewed.

I added this new page to the use cases section, which is now a group, but, of course, feel free to move it around as you see fit.

### CODE REVIEW

File: src/manifest.json [hunk 1]
```diff
@@ -8,14 +8,34 @@
...
+          "id": "redis_use_cases_basic",
+          "label": "Basic Redis use cases",
+          "summary": "A collection of common use cases for Redis.",
+          "args": {
+            "initialIsOpen": true,
+            "path": "/uc/intro.md"
+          }
+        },
+        {
+          "type": "internal-link",
```
Comment (ViktarStarastsenka) [inline src/manifest.json:29]: I think it makes sense to move this tutorial under the existing "Vector search explained" section and rename it to "Vector search and RAG".

File: src/uc/rag.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+1. **High performance and scalability**: Redis is known for its low latency and high throughput, which are essential for real-time applications. Its in-memory data store ensures quick access to data, making it ideal for applications requiring rapid data retrieval and generation.
+
+1. **Spring AI and Redis**: Using Spring AI with Redis simplifies the process of building RAG applications. Spring AI provides a structured approach to integrating AI capabilities into applications, while Redis handles the data management aspect, ensuring that the RAG pipeline is efficient and scalable.
+
+### Building a RAG application with Redis
+
+To build a RAG application with Redis, the following are some general steps:
+
+1. **Set up Redis**: Start by setting up a Redis instance and configuring it to handle vector data. The RedisVL library will be instrumental here, as it provides the necessary tools for storing and querying vector embeddings.
```
Comment (ViktarStarastsenka) [inline src/uc/rag.md:30]: I think we can promote Redis Cloud here as well
though not sure that a free Redis database will be of use here due to its size

File: src/uc/rag.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+1. **Embed and store data**: Convert your data into vector embeddings using a suitable model (e.g., BERT, GPT). Store these embeddings in Redis, where they can be quickly retrieved based on vector searches.
+
+1. **Integrate with a generative model**: Use a generative AI model that can leverage the retrieved data. The model will use the vectors stored in Redis to augment its generation process, ensuring that the output is informed by relevant, up-to-date information.
+
+1. **Query and generate**: Implement the query logic that retrieves relevant vectors from Redis based on the input prompt. Feed these vectors into the generative model to produce augmented outputs.
+
+### Benefits of Using Redis for RAG
```
Comment (ViktarStarastsenka) [inline src/uc/rag.md:38]: let's move this section above the "Building a RAG application with Redis"?
I think it is beneficial to read about benefits first

File: src/uc/rag.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+- **Efficiency**: The in-memory data store of Redis ensures that retrieval operations are performed with minimal latency, which is crucial for real-time applications.
+- **Scalability**: Redis can handle large volumes of data and scale horizontally, making it suitable for applications with growing data needs.
+- **Flexibility**: The support for various data structures and integration with different AI frameworks in Redis allows for flexible and adaptable RAG pipelines.
+
+In summary, Redis offers a powerful and efficient platform for implementing Retrieval Augmented Generation. Its vector management capabilities, high performance, and seamless integration with AI frameworks make it an ideal choice if you are looking to enhance your generative AI applications with real-time data retrieval.
+
+### Resources
+
+- [RAG defined](https://redis.io/glossary/retrieval-augmented-generation/).
```
Comment (ViktarStarastsenka) [inline src/uc/rag.md:48]: we can use the standard UTM parameters in these links

File: src/uc/rag.md (hunk 2 above)
Comment (ViktarStarastsenka) [inline src/uc/rag.md:30]: also, let's mention vector search in the first point?

**Review by ViktarStarastsenka — COMMENTED** [review]
Left some comments

File: src/uc/rag.md (hunk 4 above)
Comment (dwdougherty) [reply src/uc/rag.md:48]: And those are?

File: src/manifest.json (hunk 1 above)
Comment (dwdougherty) [reply src/manifest.json:29]: Olga wanted it at the top under the general use cases section. So, maybe check with her?

File: src/uc/rag.md (hunk 4 above)
Comment (dwdougherty) [reply src/uc/rag.md:48]: All the links?

File: src/uc/rag.md (hunk 2 above)
Comment (dwdougherty) [reply src/uc/rag.md:30]: Done

File: src/uc/rag.md (hunk 3 above)
Comment (dwdougherty) [reply src/uc/rag.md:38]: Done

File: src/uc/rag.md (hunk 2 above)
Comment (dwdougherty) [reply src/uc/rag.md:30]: Done

File: src/manifest.json (hunk 1 above)
Comment (ViktarStarastsenka) [reply src/manifest.json:29]: @ollopaci , any reason for this?

Comment (dwdougherty) [reply src/manifest.json:29]: Hi @ViktarStarastsenka and @ollopaci . I don't really have an axe to grind on how the tutorial content is organized, so I'll leave it up to you to decide.

**Review by ViktarStarastsenka — APPROVED** [review]
LGTM

**Pull request merged**